OME-418

vkhadyka opened the pull request:
Enables tap gesture recognition on the week view and provides the start date, end date, and column index of the tap location to the delegate.

This allows the delegate to accurately determine the time and resource associated with a tap, facilitating actions like event creation or selection.
Also fixes an issue where the tap gesture was not correctly enabled.

vkhadyka committed:
Adds tap gesture support with date and column info

Enables tap gesture recognition on the week view and provides the start date, end date, and column index of the tap location to the delegate.

This allows the delegate to accurately determine the time and resource associated with a tap, facilitating actions like event creation or selection.
Also fixes an issue where the tap gesture was not correctly enabled.

kvyatkovskys commented on JZCalendarWeekView/JZLongPressWeekView.swift:
```diff
@@ -269,7 +283,11 @@ open class JZLongPressWeekView: JZBaseWeekView, UIGestureRecognizerDelegate {
-    public var isTapGestureEnabled: Bool = false
+    public var isTapGestureEnabled: Bool = false {
+        didSet {
+            setupTapGestures()
```
We need to remove and readd gestures

vkhadyka committed:
Ensures tap gestures are properly set up

Corrects the order of operations to ensure that tap gesture recognizers are removed before potentially being re-added.
This prevents potential issues related to gesture recognizer setup.
